server/ladders-local.ts

```diff
@@ -15,59 +15,57 @@
 
 'use strict';
 
-/** @type {typeof import('../lib/fs').FS} */
-const FS = require(/** @type {any} */('../.lib-dist/fs')).FS;
+import {FS} from '../lib/fs';
 
 // ladderCaches = {formatid: ladder OR Promise(ladder)}
 // Use Ladders(formatid).ladder to guarantee a Promise(ladder).
 // ladder is basically a 2D array representing the corresponding ladder.tsv
 //   with userid in front
-/** @typedef {[string, number, string, number, number, number, string]} LadderRow [userid, elo, username, w, l, t, lastUpdate] */
-/** @type {Map<string, LadderRow[] | Promise<LadderRow[]>>} formatid: ladder */
-let ladderCaches = new Map();
+type LadderRow = /** [userid, elo, username, w, l, t, lastUpdate */
+[string, number, string, number, number, number, string];
+type Cache = Map<string, LadderRow[] | Promise<LadderRow[]>>;
+
+const ladderCaches: Cache = new Map();
 
 class LadderStore {
-	/**
-	 * @param {string} formatid
-	 */
-	constructor(formatid) {
+	formatid: string;
+	ladder: LadderRow[] | null;
+	ladderPromise: Promise<LadderRow[]> | null;
+	saving: boolean;
+	static formatsListPrefix: string;
+	static ladderCaches: Cache;
+
+	constructor(formatid: string) {
 		this.formatid = formatid;
-		/** @type {LadderRow[]?} */
 		this.ladder = null;
-		/** @type {Promise<LadderRow[]>?} */
 		this.ladderPromise = null;
+		this.saving = false;
 	}
 
-	/**
-	 * @return {Promise<LadderRow[]>}
-	 */
-	getLadder() {
+	getLadder(): Promise<LadderRow[]> {
```

Comment:
PS by convention doesn't specify return type except when it can't be inferred. I'd prefer to keep that consistent.

Reply:
(You specify all return types in this file, which is mostly unnecessary.)

Reply:
Hopefully https://github.com/Zarel/Pokemon-Showdown/pull/5490/commits/365868d02ab1c6addd50bf147362a63f21a6f7a5 fixed what you mean?

Reply:
Yep, that's what I wanted; thanks!